cms/app_registration.py

```diff
@@ -1,27 +1,96 @@
+import inspect
 from importlib import import_module
 
 from django.apps import apps
 from django.utils.module_loading import module_has_submodule
+from django.core.exceptions import ImproperlyConfigured
 
 
 CMS_CONFIG_NAME = 'cms_apps'
 
 
-def autodiscover_cms_files():
-    """Find and import all cms_apps.py modules"""
+def autodiscover_cms_configs():
+    """
+    Find and import all cms_apps.py files. Add a cms_app attribute
+    to django's app config with an instance of the cms config.
+    """
     for app_config in apps.get_app_configs():
         try:
-            import_module('%s.%s' % (app_config.name, CMS_CONFIG_NAME))
-        except Exception:
+            cms_module = import_module(
+                '%s.%s' % (app_config.name, CMS_CONFIG_NAME))
+        except:
+            # If something in cms_apps.py raises an exception let that
+            # exception bubble up. Only catch the exception if
+            # cms_apps.py doesn't exist
             if module_has_submodule(app_config.module, CMS_CONFIG_NAME):
```

Comment:
Is this bubble up exception tested? I couldn't see a test that mentioned this.

Reply:
https://github.com/monikasulik/django-cms/blob/b5f1825e1225e00de25f6fe4114a512c04c18708/cms/tests/test_app_registration.py#L54
:smile: